lib/evse_security.cpp

```diff
@@ -329,22 +372,29 @@ EvseSecurity::get_installed_certificates(const std::vector<CertificateType>& cer
         auto ca_bundle_path = this->ca_bundle_path_map.at(ca_certificate_type);
         try {
             X509CertificateBundle ca_bundle(ca_bundle_path, EncodingFormat::PEM);
-            auto certificates_of_bundle = ca_bundle.split();
+            X509CertificateHierarchy& hierarchy = ca_bundle.get_certficate_hierarchy();
 
-            CertificateHashDataChain certificate_hash_data_chain;
-            certificate_hash_data_chain.certificate_type =
-                get_certificate_type(ca_certificate_type); // We always know type
+            EVLOG_info << "Hierarchy:(" << conversions::ca_certificate_type_to_string(ca_certificate_type) << ")\n"
+                       << hierarchy.to_debug_string();
 
-            for (int i = 0; i < certificates_of_bundle.size(); i++) {
-                CertificateHashData certificate_hash_data = certificates_of_bundle[i].get_certificate_hash_data();
-                if (i == 0) {
-                    certificate_hash_data_chain.certificate_hash_data = certificate_hash_data;
-                } else {
-                    certificate_hash_data_chain.child_certificate_hash_data.push_back(certificate_hash_data);
-                }
-            }
+            // Iterate the hierarchy and add all the certificates to their respective locations
+            for (auto& root : hierarchy.get_hierarchy()) {
+                CertificateHashDataChain certificate_hash_data_chain;
+
+                certificate_hash_data_chain.certificate_type =
+                    get_certificate_type(ca_certificate_type); // We always know type
+                certificate_hash_data_chain.certificate_hash_data = root.hash;
+
+                // Add all owned children/certificates in order
+                X509CertificateHierarchy::for_each_child(
+                    [&certificate_hash_data_chain](const X509Node& child, int depth) {
+                        certificate_hash_data_chain.child_certificate_hash_data.push_back(child.hash);
+                    },
+                    root);
```

Comment:
This will add not just the direct children, but all ancestors of the given root certificate. Is this what we intend to happen here?

Additionally, it will only find the children within the same bundle, but not those which reside in another bundle.

Reply:
Tracking children across multiple bundles would require building a hierarchy out of the whole certificate store. It might be added in a future PR.